File: sample-rest-api/src/main/java/nl/haploid/octowight/sample/service/CaptainService.java
package nl.haploid.octowight.sample.service;

import nl.haploid.octowight.sample.data.CaptainModel;
import nl.haploid.octowight.sample.data.CaptainResource;
import org.springframework.stereotype.Service;

@Service
public class CaptainService extends AbstractResourceService<CaptainModel, CaptainResource> {

	@Override
	protected Class<CaptainModel> getModelClass() {
		return CaptainModel.class;
	}

	@Override
	public String getResourceType() {
		return CaptainResource.ResourceType();
	}
}

File: sample-rest-api/src/main/scala/nl/haploid/octowight/sample/service/CaptainService.scala
package nl.haploid.octowight.sample.service

import nl.haploid.octowight.sample.data.{CaptainModel, CaptainResource}
import org.springframework.stereotype.Service

@Service
class CaptainService extends AbstractResourceService[CaptainModel, CaptainResource] {

  def modelClass = classOf[CaptainModel]

  def collection = CaptainResource.ResourceCollection
}
